### PR TITLE
Updated Default values table

### PR DESCRIPTION
**Pull request by pkulikov**
This PR updates the [Default values table](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/default-values-table) article in the following way:
- Move the table to the top of the article, all other content - to the **Remarks** section. To be consistent with other table articles.
- Focus the **Remarks** section on *how* to produce the default value: which is either the default value expression or the default constructor of a value type.
- Mention default values of reference and nullable types for completeness. 

I didn't touch the `helpviewer_keywords` section in metadata. Please let me know if it should be updated as well.

### CODE REVIEW

File: docs/csharp/language-reference/keywords/default-values-table.md [hunk 1]
```diff
@@ -46,10 +34,34 @@ Remember that using uninitialized variables in C# is not allowed.
...
+int a = default(int);
+```
+
+Beginning with C# 7.1, you can use the [`default` literal](../../programming-guide/statements-expressions-operators/default-value-expressions.md#default-literal-and-type-inference) to initialize a variable with the default value of its type:
+
+```csharp
+int a = default;
+```
+
+You also can use the default constructor to produce the default value of a value type:
```
Comment (rpetrusha) [inline docs/csharp/language-reference/keywords/default-values-table.md]: This is unrelated to your PR, but I'd prefer the following change:

default constructor --> default constructor or the implicit default constructor

The problem here is that the primitives don't actually have constructors; they're an artifact of the compiler, which replaces the constructor call with a call to the IL Initobj instruction. If you want, you can cross-reference the [constructors topic](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/constructors), which mentions both default and implicit default constructors, or an [old blog post on implicit default constructors](https://blogs.msdn.microsoft.com/bclteam/2010/04/28/constructors-and-value-types-ron-petrusha/).

**Review by rpetrusha — CHANGES_REQUESTED** [review]
This is a vast improvement, @pkulikov; thank you so much for working on this topic. I've left a couple of comments.

**Comment by pkulikov** [comment]
@rpetrusha thank you for review; I've addressed  your feedback (also described one more example where default values might be useful: default values of method optional arguments).

File: docs/csharp/language-reference/keywords/default-values-table.md [hunk 2]
```diff
@@ -46,10 +34,36 @@ Remember that using uninitialized variables in C# is not allowed.
 |[ulong](ulong.md)|0|
 |[ushort](ushort.md)|0|
 
+## Remarks
+
+You cannot use uninitialized variables in C#. You can initialize a variable with the default value of its type. You also can use the default value of a type to specify the default value of a method's [optional argument](../../programming-guide/classes-and-structs/named-and-optional-arguments.md#optional-arguments).
+
+Use [default value expression](../../programming-guide/statements-expressions-operators/default-value-expressions.md) to produce the default value of a type as the following example shows:
```
Comment (rpetrusha) [inline docs/csharp/language-reference/keywords/default-values-table.md]: two nits that I missed in my earlier review:
- Use --> Use the
- comma before "as"

**Review by rpetrusha — CHANGES_REQUESTED** [review]
I apologize, @pkulikov -- I missed two nits when I reviewed your PR earlier. It would be nice to fix them before we merge. Otherwise, this looks great.

**Comment by pkulikov** [comment]
@rpetrusha no worries, I addressed your feedback

**Review by rpetrusha — APPROVED** [review]
Thanks, @pkulikov. This looks great. I'll merge now.

**Pull request merged**